src/pages/Social/Social.js

```diff
@@ -0,0 +1,59 @@
+import React, { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import styled from 'styled-components';
+import { CLIENT_ID, REDIRECT_URI, BASE_URL } from '../../config';
+
+const Social = () => {
+  const [token, setToken] = useState('');
+
+  const navigate = useNavigate();
+
+  const code = new URL(window.location.href).searchParams.get('code');
+
+  useEffect(() => {
+    fetch(`https://kauth.kakao.com/oauth/token`, {
+      method: 'POST',
+      headers: {
+        'Content-type': 'application/x-www-form-urlencoded;charset=utf-8',
+      },
+      body: `grant_type=authorization_code&client_id=${CLIENT_ID}&code=${code}&redirect_uri=${REDIRECT_URI}`,
+    })
+      .then(res => res.json())
+      .then(data => setToken(data.access_token));
+  }, []);
+
+  useEffect(() => {
+    token &&
+      fetch(`${BASE_URL}/users/kakaosignin`, {
+        headers: {
+          Authorization: token,
+        },
+      })
+        .then(res => res.json())
+        .then(data => localStorage.setItem('token', data.access_token))
+        .then(navigate('/'));
+  }, [token]);
```

Comment:
```suggestion
        .then(res => res.json())
        .then(data => {
        localStorage.setItem('token', data.access_token)
        navigate('/')}
        )
  }, [token]);
```
* 이 응답도, 통신 성공 실패에 따른 key를 확인 해 보시고, 응답에 따른 분기 처리로 해주세요!
* navigate는 토큰 저장후에 바로 호출하면 됩니다!